src/fsharp/fsi/fsi.fs

```diff
@@ -1382,6 +1395,28 @@ type internal FsiDynamicCompiler
           let istate = (istate, sourceFiles, inputs) |||> List.fold2 (fun istate sourceFile input -> fsiDynamicCompiler.ProcessMetaCommandsFromInputAsInteractiveCommands(ctok, istate, sourceFile, input))
           fsiDynamicCompiler.EvalParsedSourceFiles (ctok, errorLogger, istate, inputs)
 
+    member __.GetBoundValues istate =
+        let cenv = SymbolEnv(istate.tcGlobals, istate.tcState.Ccu, Some istate.tcState.CcuSig, istate.tcImports)
+        [ for pair in istate.boundValues do
+            let nm = pair.Key
+            let v = pair.Value
+            match tryGetGeneratedValue istate cenv v with
+            | Some fsiValue ->
+                yield FsiBoundValue(nm, fsiValue)
+            | _ ->
```

Comment:
When would `istate.boundValues` have an item, only to not actually be found by `tryGetGeneratedValue`?

Reply:
I think I see what you are saying. Why would there be a `try` in the first place? If it's there, then we don't need to `try`. I can always see `tryGetGeneratedValue` returning something for each item in `boundValues`. We could throw an error instead on of returning nothing, but I was following the existing pattern of handling results when looking up the generated values.